internal/server/workflow_vars.go

```diff
@@ -119,6 +126,25 @@ func getSelectedTeam(r *http.Request, loggedInTeamId int, defaultTeamId int, tea
 	return model.Team{}, errors.New("invalid team selection")
 }
 
+func createReassignDeputyDropDownList(allTeams []model.Team, requiredTeamTypes []string, currentSelectedTeam model.Team) []model.Team {
```

Comment:
As above, this should be a function of the DeputiesPage struct.

I'd also suggest not referring to the dropdown list in the function name. That's what it is being used for, not what it is doing. If we wanted to use this filtered team list for another purpose, we would need to rename it or it would be confusing. Say what it is doing, not why (e.g. `filterTeamsByTypes`).